### PR TITLE
Skip sig-verify for all-zero signature in Solana compiler.

### PR DESCRIPTION
**Pull request by damian-zhao**
## Description

Related issue: https://github.com/trustwallet/wallet-core/issues/3749

## How to test

Run the newly added unit tests in `solana_compile.rs`

## Types of changes

Enhancement

## Checklist

- [ ] Create pull request as draft initially, unless its complete.
- [x] Add tests to cover changes as needed.
- [ ] Update documentation as needed.
- [x] If there is a related Issue, mention it in the description.

### CODE REVIEW

File: rust/chains/tw_solana/src/compiler.rs [hunk 1]
```diff
@@ -86,7 +89,8 @@ impl SolanaCompiler {
             let signature = ed25519::Signature::try_from(sign.as_slice())?;
             let pubkey = ed25519::sha512::PublicKey::try_from(pubkey.as_slice())?;
 
-            if !pubkey.verify(signature.clone(), data_to_sign.clone()) {
+            if !pubkey.verify(signature.clone(), data_to_sign.clone())
+                && signature.to_bytes() != H512::from(ZERO_SIGNATURE) {
```
Comment (satoshiotomakan) [inline rust/chains/tw_solana/src/compiler.rs:93]: It's not optimal to parse the string as H512 on each iteration.
Could you please add `Hash<N>::is_zero`:
```rust
pub fn is_zero(&self) -> bool {
    self.0.iter().all(|byte| *byte == 0)
}
```

and use it this way:
```rust
if !pubkey.verify(signature.clone(), data_to_sign.clone()) && !signature.to_bytes().is_zero() {
    return Err();
}
```

**Review by satoshiotomakan — CHANGES_REQUESTED** [review]
Thank you for adding tests! Could you consider one minor change?

File: rust/chains/tw_solana/src/compiler.rs (hunk 1 above)
Comment (damian-zhao) [reply rust/chains/tw_solana/src/compiler.rs]: Sure, and done.

**Review by satoshiotomakan — DISMISSED** [review]
Thank you

**Comment by damian-zhao** [comment]
Hi @ar-g , do you know why Kotlin Multiplatform CI failed?

**Comment by satoshiotomakan** [comment]
@damian-zhao the KMP CI should be fixed when 4.0.31 release is ready. Will update this branch after we have a 4.0.31 release available

**Pull request merged**